File: server/internal/model/article.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Article struct {
	ID          uint64         `gorm:"primaryKey" json:"id"`
	Title       string         `gorm:"size:180;not null" json:"title"`
	Slug        string         `gorm:"size:220;not null;uniqueIndex" json:"slug"`
	Summary     string         `gorm:"size:500" json:"summary"`
	ContentMD   string         `gorm:"type:longtext" json:"content_md"`
	Status      string         `gorm:"size:32;not null;index" json:"status"`
	TopicID     uint64         `gorm:"index" json:"topic_id"`
	Topic       Topic          `json:"topic_detail"`
	AuthorID    uint64         `gorm:"index" json:"author_id"`
	Author      User           `json:"author_detail"`
	Tags        []Tag          `gorm:"many2many:article_tags;" json:"tag_details"`
	ViewCount   uint64         `gorm:"not null;default:0" json:"view_count"`
	PublishedAt *time.Time     `gorm:"index" json:"published_at"`
	CreatedAt   time.Time      `gorm:"index" json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

File: web/src/types/article.ts
export interface ArticleTopic {
  id: number
  name: string
  label: string
  slug: string
}

export interface ArticleListItem {
  id: number
  title: string
  slug: string
  summary: string
  cover_url: string
  status: 'draft' | 'published' | 'private' | 'archived'
  topic_id: number
  topic: ArticleTopic
  tags: string[]
  view_count: number
  published_at: string
  created_at: string
  updated_at: string
}

export interface ArticleDetail extends ArticleListItem {
  content_md: string
  tag_ids: number[]
}

export interface ArticleSearchItem extends ArticleListItem {
  snippet: string
  matched_fields: string[]
}

export interface ArticleVersionItem {
  id: number
  article_id: number
  title: string
  summary: string
  content_md: string
  status: ArticleListItem['status']
  created_at: string
}

export interface ArticleSavePayload {
  title: string
  slug: string
  summary: string
  cover_url: string
  content_md: string
  topic_id: number
  tag_ids: number[]
  status: ArticleListItem['status']
}
